iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerNoScan.java

```diff
@@ -1052,6 +1053,47 @@ public void testDropTableWithAppendedData() throws IOException {
     shell.executeStatement("DROP TABLE customers");
   }
 
+  @Test
+  public void testDropTableWithoutPurge() throws IOException {
+    TableIdentifier identifier = TableIdentifier.of("default", "customers");
+
+    shell.executeStatement("CREATE EXTERNAL TABLE customers (" +
+        "t_int INT,  " +
+        "t_string STRING) " +
+        "STORED BY ICEBERG " +
+        testTables.locationForCreateTableSQL(identifier) +
+        testTables.propertiesForCreateTableSQL(ImmutableMap.of(InputFormatConfig.EXTERNAL_TABLE_PURGE, "FALSE")));
```

Comment:
How about running this test without adding the purge=false flag explicitly? Explicitly adding it with FALSE will circumvent the MetaHook behaviour for the default case where the purge flag is simply omitted